cfme/automate/explorer/domain.py

```diff
@@ -0,0 +1,372 @@
+        for domain in checked_domains:
+            all_page.flash.assert_message(
+                'Automate Domain "{}": Delete successful'.format(domain.description or domain.name))
+
+    def set_order(self, items):
+        if not isinstance(items, (list, tuple)):
+            items = [items]
+
+        processed_items = [Fillable.coerce(item) for item in items]
+        priority_page = navigate_to(self, 'Priority')
+        changed = priority_page.domains.fill(processed_items)
+        if changed:
+            # Changed
+            priority_page.save_button.click()
+        else:
+            # Not changed
+            priority_page.cancel_button.click()
+        domains_view = self.create_view(DomainListView)
+        assert domains_view.is_displayed
+        domains_view.flash.assert_no_error()
+        if changed:
+            domains_view.flash.assert_message('Priority Order was saved')
+        else:
+            domains_view.flash.assert_message('Edit of Priority Order was cancelled by the user')
+        return changed
+
+
+@navigator.register(DomainCollection)
+class All(CFMENavigateStep):
+    VIEW = DomainListView
+    prerequisite = NavigateToAttribute('appliance.server', 'AutomateExplorer')
+
+    def step(self):
+        self.prerequisite_view.datastore.tree.click_path(*self.obj.tree_path)
+
+
+@navigator.register(DomainCollection)
+class Add(CFMENavigateStep):
+    VIEW = DomainAddView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select('Add a New Domain')
+
+
+@navigator.register(DomainCollection)
+class Priority(CFMENavigateStep):
+    VIEW = DomainPriorityView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select('Edit Priority Order of Domains')
+
+
+class DomainDetailsView(AutomateExplorerView):
+    title = Text('#explorer_title_text')
+    namespaces = Table('#ns_details_grid')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and
+            re.match(
+                r'^Automate Domain "{}(?: \((?:Disabled|Locked|Locked & Disabled)\))?"$'.format(
+                    re.escape(self.context['object'].name)),
+                self.title.text) is not None)
+
+
+class Domain(Navigatable, Fillable):
+    """A class representing one Domain in the UI."""
+    def __init__(
+            self, name, description, enabled=None, locked=None, collection=None, appliance=None):
+        if collection is None:
+            collection = DomainCollection(appliance=appliance)
+        self.collection = collection
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.name = name
+        self.description = description
+        self.enabled = enabled
+        self.locked = locked
+        self._db_id = None
+
+    def as_fill_value(self):
+        return self.name
+
+    @cached_property
+    def db_id(self):
```

Comment:
this looks like this cached property was accidentally implemented like a property that caches would have been
is this a rebase artifact?